### ISSUE TITLE
How was PRIESSTESS run with LARP6?

### ISSUE THREAD

**Issue by byee4** [kaitlin309/PRIESSTESS#7]
Hi,

Sorry, I just saw that the Benchmark csv file describes which datasets were used for SLBP. So I'll have to amend my question to ask if there is any data related to the other RBP discussed in the paper (LARP6)?

**Comment by byee4** [comment]
Oops found it. Didn't see that it was uploaded here, my mistake:

https://www.ncbi.nlm.nih.gov/biosample/29781374
https://www.ncbi.nlm.nih.gov/biosample/29781375
https://www.ncbi.nlm.nih.gov/biosample/29781376
https://www.ncbi.nlm.nih.gov/biosample/29781377

**Comment by kaitlin309** [comment]
Hi Brian,

Yes - that is correct, I didn't run it on the eCLIP. However, you could alter the code in the man PRIESSTESS file if you want run it on that. I can help you out if you have any issues.

Also, let me know if you end up having any issues with running it on SLBP and LARP6 HTR-SELEX/RNAcompete-S data.

Kailtin

**Comment by byee4** [comment]
Hi Kaitlin, thanks! I believe I was able to run the pipeline, and am now trying to re-create the figures (4 and 5) using the data to ensure we have things working. However, I'm unsure what files would need to be used or would be considered the actual output. It looks like fg_alphabet_annotations.tab encodes the sequence alphabets for each binding site/peak, and that Figure 5 simply plot the top 10 scores in test_data_scores.tab for each of the five RBPs?

```
../../../../../PRIESSTESS_output/test_test_data/
├── bg_alphabet_annotations.tab
├── fg_alphabet_annotations.tab
├── seq-4
│   ├── bg_test.fa
│   ├── bg_test_PFM_scan_sum_top_4.tab
│   ├── fg_test.fa
│   └── fg_test_PFM_scan_sum_top_4.tab
├── seq-struct-16
│   ├── bg_test.fa
│   ├── bg_test_PFM_scan_sum_top_4.tab
│   ├── fg_test.fa
│   └── fg_test_PFM_scan_sum_top_4.tab
├── seq-struct-28
│   ├── bg_test.fa
│   ├── bg_test_PFM_scan_sum_top_4.tab
│   ├── fg_test.fa
│   └── fg_test_PFM_scan_sum_top_4.tab
├── seq-struct-8
│   ├── bg_test.fa
│   ├── bg_test_PFM_scan_sum_top_4.tab
│   ├── fg_test.fa
│   └── fg_test_PFM_scan_sum_top_4.tab
├── struct-2
│   ├── bg_test.fa
│   ├── bg_test_PFM_scan_sum_top_4.tab
│   ├── fg_test.fa
│   └── fg_test_PFM_scan_sum_top_4.tab
├── struct-4
│   ├── bg_test.fa
│   ├── bg_test_PFM_scan_sum_top_4.tab
│   ├── fg_test.fa
│   └── fg_test_PFM_scan_sum_top_4.tab
├── struct-7
│   ├── bg_test.fa
│   ├── bg_test_PFM_scan_sum_top_4.tab
│   ├── fg_test.fa
│   └── fg_test_PFM_scan_sum_top_4.tab
├── test_data_scores.tab
└── test_PRIESSTESS_model_ON_test_data_auroc.tab

```

**Comment by kaitlin309** [comment]
Hi Brian, 

It looks like that is a directory resulting from running `PRIESSTESS_scan`, is that correct? Figures 4 and 5 don't use any results from testing on other datasets - they just plot information derived from the trained models. 

I am going to clean up my plotting scripts and upload them so that you can recreate the figures exactly and see what they are plotting.

**Comment by byee4** [comment]
Oh, sorry I may be confused then. Yes that would be super helpful, thanks! It would be good to know/confirm what the outputs mean as well.

**Comment by kaitlin309** [comment]
Here is a description of all the files resulting from `PRIESTESS_scan`:

Sequences from the input positive and negative test sets annotated with all 7 alphabets:  `fg_alphabet_annotations.tab` & `bg_alphabet_annotations.tab`

For each alphabet, a directory containing:

FASTA files containing only the corresponding annotation for positive and negative test sets: `fg_test.fa` & `bg_test.fa`
TAB delimited files containing the PFM scores on the positive and negative test set sequences: `fg_test_PFM_scan_sum_top_4.tab` & `bg_test_PFM_scan_sum_top_4.tab`

In these two TAB-delimited files there is a column for each PFM used in the PRIESSTESS model that is being applied to the test data. For example, call the PRIESSTESS model being applied to this dataset PX. If four seq-struct-28 PFMs were identified by STREME in PX, the `fg_test_PFM_scan_sum_top_4.tab` & `bg_test_PFM_scan_sum_top_4.tab` files in the `seq-struct-28` directory will contain a column for each of the four PFMs with the score of those PFMs on each of the test sequences.

`test_data_scores.tab` is created by performing the following steps:
- concatenating the `fg_test_PFM_scan_sum_top_4.tab` files from all alphabets column-wise
- concatenating the `bg_test_PFM_scan_sum_top_4.tab` files from all alphabets column-wise
- concatenating the files created by the previous two steps row-wise and adding a column with the class label (bg/negative - 0, fg/positive - 1)

This file has the same format as the training data for the original PRIESSTESS model that is being applied and is used in the script `test_PRIESSTESS_model.py`. The sequences are then scored by the logistic regression model to produce an AUROC which is held in `test_PRIESSTESS_model_ON_test_data_auroc.tab`.

However, I can see that some users would probably like not only the AUROC but the individual scores of the model on the test sequences. I will update `test_PRIESSTESS_model.py` to produce a file containing this.

**Comment by kaitlin309** [comment]
Just to update you: the plotting scripts will be in PRIESSTESS/plotting_scripts
